### DISCUSSION TITLE
Read a config from ".github" repository

### DISCUSSION THREAD

**Discussion by binyamin** [renovatebot/renovate#12383] [Request Help] [answered]
### What would you like Renovate to be able to do?

**Problem**
With the GitHub-hosted app, it sounds like (#11262) you can't create a default `renovate.json` for an account/organization.

**Suggestion**
What if you could use a `renovate.json` file in a ".github" repo? GitHub already uses that system for [some default configs](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/creating-a-default-community-health-file#supported-file-types)), like contributing guidelines.

**Benefits**
1. Obviously, it permits setting a default configuration for the GitHub-hosted app.
2. This would allow users to bypass the initial PR when enabling Renovate for another repository, simplifying the process.
3. A maintainer might want to separate GitHub-specific configuration from the source-code.

### If you have any ideas on how this should be implemented, please tell us here.

___Disclaimer__: I'm not super-familiar with the source-code, but here's my take on implementation._

I'm aware of at least three possible scenarios here:
1. The default config exists, and the user enables/creates a repository.
2. The default config exists, and the user deletes a repository-specific config.
3. The user enables or creates a repository, so Renovate creates an initial setup PR. Then, the user creates a default config, meaning the setup PR may not be necessary.

I think the safest and most effective solution, would be do try loading the default config whenever a local config doesn't exist. This definitely solves scenario 2. Maybe scenario 1, depending on the code. It would solve scenario 3; however, the user would need to manually tell Renovate to retry each existing setup PR.

Scenario 1 might also work by modifying [`/lib/workers/repository/init/*.ts`](https://github.com/renovatebot/renovate/tree/main/lib/workers/repository/init).

For scenario 3, Renovate could also close all setup PRs when it detects a config file in the ".github" repository (or when it changes). Then the user wouldn't need to deal with each setup PR individually.

### Is this a feature you are interested in implementing yourself?

No (*Edit*: I'm not opposed to patching this; I just don't want to make a mess of the code)

**Comment by viceice** [comment, marked as answer]
It's already possible. For an automatic org wide onboarding config, you should use a `/defauld.json`, the `/renovate.json` is deprecated but still works, so you should use `.github/renovate.json` file for the local config.

Checkout our `renovatebot/.github` repo for sample.

**Reply by binyamin** [reply]
The name `default.json`, is very vague. Is that the only option?

**Reply by binyamin** [reply]
Also, shouldn't this be in the docs?

**Reply by viceice** [reply]
https://docs.renovatebot.com/config-presets/

**Reply by rarkins** [reply]
Specifically "Organization level presets"
